app/workers/new_page_worker.rb

```diff
@@ -0,0 +1,44 @@
+class NewPageWorker
+  include Sidekiq::Worker
+  attr_reader :page, :doc_html
+
+  def perform(page_id)
+    @page = Page.find(page_id)
+    @doc_html = download_content
+    process
+  rescue SocketError, Gastly::FetchError, OpenURI::HTTPError, Net::OpenTimeout
+    page.update_attributes(status: Page::ERROR_STATUS)
+  end
+
+  private
+
+  def process
+    page.update_attributes(status: Page::IN_PROGRESS_STATUS)
+    parse_html_content
+    make_screenshot
+    page.update_attributes(status: Page::PROCESSED_STATUS)
+  end
+
+  def download_screenshot(file)
+    File.new(Gastly.capture(page.source_url, file.path))
+  end
+
+  def make_screenshot
+    file = Tempfile.new
+    page.update_attributes(screenshot: download_screenshot(file))
+    file.close!
```

Comment:
where is unlink?
also it must be added like

```
def xxx
 f = Temfile.new
ensure
 f.close
 f.unlink
end
```

Reply:
close!() click to toggle source
Closes and unlinks (deletes) the file. Has the same effect as called close(true).

Reply:
yes, but add "ensure" .... in case of some exception - file ALWAYS will be closed

Reply:
✓Done